pkg/compiler/compiler.go

```diff
@@ -322,11 +322,15 @@ func (cs *LuetCompiler) buildPackageImage(image, buildertaggedImage, packageImag
 	// so the hash is unique also in cases where
 	// some package deps does have completely different
 	// depgraphs
-	// TODO: As the salt contains the packageImage ( in registry/organization/imagename:tag format)
-	// the images hashes are broken with registry mirrors.
-	// We should use the image tag, or pass by the package assertion hash which is unique
+	// TODO: We should use the image tag, or pass by the package assertion hash which is unique
 	// and identifies the deptree of the package.
 
+	imgRef := strings.SplitN(packageImage, "/", 2)
+	if len(imgRef) == 2 && helpers.IsDomainName(imgRef[0]) {
```

Comment:
better to move the whole as a function, and give the result back to  `HashFingerPrint`, for example:

```golang
fp := p.GetPackage().HashFingerprint(StripRegistryFromImage(packageImage))
```

In this way we can also write unit tests for `StripRegistryFromImage` easily :slightly_smiling_face: